sdk/iot/core/inc/az_iot_hub_client.h

```diff
@@ -0,0 +1,505 @@
+ */
+AZ_NODISCARD az_result az_iot_hub_client_id_get(
+    az_iot_hub_client const* client,
+    az_span mqtt_client_id,
+    az_span* out_mqtt_client_id);
+
+/**
+ *
+ * SAS Token APIs
+ *
+ *   Use the following APIs when the Shared Access Key is available to the application or stored
+ *   within a Hardware Security Module. The APIs are not necessary if X509 Client Certificate
+ *   Authentication is used.
+ */
+
+/**
+ * @brief Gets the Shared Access clear-text signature.
+ * @details The application must obtain a valid clear-text signature using
+ *          this API, sign it using HMAC-SHA256 using the Shared Access Key as password then Base64
+ *          encode the result.
+ *
+ * @param[in] client
+ * @param[in] token_expiration_unix_time The time, in seconds, from the Unix epoch (1/1/1970).
+ * @param[in] signature An empty #az_span with sufficient capacity to hold the SAS signature.
+ * @param[out] out_signature The output #az_span containing the SAS signature.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_hub_client_sas_signature_get(
+    az_iot_hub_client const* client,
+    uint32_t token_expiration_unix_time,
+    az_span signature,
+    az_span* out_signature);
+
+/**
+ * @brief Gets the MQTT password.
+ * @note The MQTT password must be an empty string if X509 Client certificates are used. Use this
+ *       API only when authenticating with SAS tokens.
+ *
+ * @param[in] client
+ * @param[in] base64_hmac_sha256_signature The Base64 encoded value of the HMAC-SHA256(signature,
+ *                                         SharedAccessKey). The signature is obtained by using
+ *                                         #az_iot_hub_client_sas_signature_get.
+ * @param[in] key_name The Shared Access Key Name (Policy Name). This is optional. For security
+ *                     reasons we recommend using one key per device instead of using a global
+ *                     policy key.
+ * @param[in] mqtt_password An empty #az_span with sufficient capacity to hold the MQTT password.
+ * @param[out] out_mqtt_password The output #az_span containing the MQTT password.
+ * @return #az_result.
+ */
+AZ_NODISCARD az_result az_iot_hub_client_sas_password_get(
+    az_iot_hub_client const* client,
+    az_span base64_hmac_sha256_signature,
+    az_span key_name,
+    az_span mqtt_password,
+    az_span* out_mqtt_password);
+
+/**
+ *
+ * Properties APIs
+ *
+ *   IoT Hub message properties are used for Device to Cloud (D2C) as well as Cloud to Device (C2D).
+ *   Properties are always appended to the MQTT topic of the published or received message and
+ *   must contain Uri-encoded keys and values.
+ */
+
+/**
+ * @brief Telemetry or C2D properties.
+ *
+ */
+typedef struct az_iot_hub_client_properties
+{
+  struct
+  {
+    az_span properties;
+    uint8_t* current_property;
+  } _internal;
+} az_iot_hub_client_properties;
+
+/**
+ * @brief Initializes the Telemetry or C2D properties.
+ *
+ * @param properties

```

Comment:
>param [](start = 4, length = 5)

Nit: param beyond this point are missing the [in]/[out] marker in doxygen.

Reply:
Yes: sorry about that - had to interrupt my work (why I've added "WIP - work in progress"). Will update all.